### ISSUE TITLE
cost error

### ISSUE THREAD

**Issue by alirezahajitabar** [ondrejklejch/ctc#2]
Dear Ondrej,
I'm trying to run your code on TIMIT dataset, first I run it its run correctly till iteration: 16  i have this ERROR:
iteration: 16
training cost: 216740.999548
training cer: 0.4090
validation cer: 0.4408 (0.4408)
epoch time: 3363.8226
Expected: sil aa r y ih l uh sil k iy n f r ow m sil p l oy m ih sil t sil
Received: sil iy f ah sil t sil
Expected: sil eh n w ah dx ay z dh ey w er sil
Received: sil m sil w sil
Expected: sil y ih dx ow n iy dx ah n ah f sil hh ah n iy sil
Received: sil ow iy n ih n hh ah n dx iy sil
Expected: sil sh iy ih z th ih n er dh ih n ay ae m sil
Received: sil iy s sil t eh n z ih n sil
Expected: sil hh aw sil g uh dx ih z y er ih n sil d er ih n s sil
Received: sil hh sil g ih dx ih z n sil d sil
16 73054.9019592 0.409018897008 0.440809537011 3363.82264304 1113.118186 37301.3046875 0.025
OVERFLOW
Total Cost:  20552.768049
Actual Cost: nan
Zero predictions:  1506
nan nan nan nan
Closing remaining open files:/home/hajitabar/proje/train.hj...done

i run it on CPU after that Any time i run it i have the same ERROR on iteration 1 or 2. i run it on GPU and i see the same ERROR.
is it relevance to theano.function?

**Comment by ondrejklejch** [comment]
Dear Alireza,

this error happens because some operation is not numerically stable. Which network are you using? It shouldn't be related to theano.function.

Best,

Ondrej

**Comment by alirezahajitabar** [comment]
thanks , I use the blstm_dropout network.

**Comment by ondrejklejch** [comment]
Can you try blstm without dropout?

**Comment by alirezahajitabar** [comment]
OK. I will try it and I will let you know the result. 
Thanks.

**Comment by alirezahajitabar** [comment]
yeah i Tried blstm network  and lstm several times and  i have the same error.what should i do to solve this problem?
Thanks

**Comment by ondrejklejch** [comment]
Dear Alireza,

as I said, there is some numerical instability which causes CTC to give infinite cost (caused by zero probability of some character), you might want to replace Log with some safe Log. I'm sorry, but I can't guide you more information.

Best

Ondre
